crates/types/src/invocation.rs

```diff
@@ -970,6 +984,205 @@ impl WithInvocationId for NotifySignalRequest {
     }
 }
 
+mod serde_hacks {
+    //! Module where we hide all the hacks to make back-compat working!
+
+    use super::*;
+
+    #[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+    pub(super) struct ServiceInvocation {
+        pub invocation_id: InvocationId,
+        pub invocation_target: InvocationTarget,
+        pub argument: Bytes,
+        pub source: Source,
+        pub span_context: ServiceInvocationSpanContext,
+        pub headers: Vec<Header>,
+        pub execution_time: Option<MillisSinceEpoch>,
+        pub completion_retention_duration: Option<Duration>,
+        pub idempotency_key: Option<ByteString>,
+        pub response_sink: Option<ServiceInvocationResponseSink>,
+        pub submit_notification_sink: Option<SubmitNotificationSink>,
+
+        // TODO(slinkydeveloper) this field is here because serde doesn't like much when I change the shape of an enum variant from empty to tuple/named fiels
+        pub source_ingress_rpc_id: Option<PartitionProcessorRpcRequestId>,
+    }
+
+    #[derive(Debug, Clone, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+    pub(super) enum Source {
+        Ingress,
```

Comment:
What's our way forward to get rid of the empty enum variant?

Reply:
We'll think about it, perhaps we introduce a new variant name, and we do the usual skip release dance to phase out this variant. Also if we remove flexbuffers moving forward, this is probably irrelevant anyway.